.github/workflows/clang-tidy.yml

```diff
@@ -39,9 +39,9 @@ jobs:
 
     env:
       CMAKE_PREFIX_PATH: /home/linuxbrew/.linuxbrew
-      CC: clang-14
-      CXX: clang++-14
-      CLANG_TIDY: clang-tidy-14
+      CC: clang-15
+      CXX: clang++-15
+      CLANG_TIDY: clang-tidy-15
```

Comment:
optional full clang-tidy run on all targets in https://github.com/PowerGridModel/power-grid-model/actions/runs/5877044919

please resolve this conversation after that build passes (see also status badge: [![Clang Tidy](https://github.com/PowerGridModel/power-grid-model/actions/workflows/clang-tidy.yml/badge.svg?branch=feature%2Fclang-tidy-run&event=workflow_dispatch)](https://github.com/PowerGridModel/power-grid-model/actions/workflows/clang-tidy.yml) )

Reply:
i did run before and that run failed, hence the failing status badge. i specifically targeted this branch for the badge

However, it turns out that there's one more remaining issue that i'll have to sort out first